src/scripts/core/axis/axis-scale-factory.js

```diff
@@ -36,11 +36,11 @@
                 axisType = 'linear';
             }
 
-            if (axisType === 'linear' && (options.yAxis.smartAxis || options.yAxis.scaling.smartAxis)) {
+            if (axisType === 'linear' && (options.yAxis.smartAxis || options.yAxis.scaling.type === 'smart')) {
                 axisType = 'smart';
             }
 
-            if (axisType === 'linear' && (options.yAxis.centeredAxis || options.yAxis.scaling.centeredAxis)) {
+            if (axisType === 'linear' && (options.yAxis.centeredAxis || options.yAxis.scaling.type === 'centered')) {
```

Comment:
I don't think we should support `. centeredAxis` -- we should still support `smartAxis` for legacy reasons, but going forward mandate everything else should be set through scaling.type.